iso8583-tcp-integration-and-message-transformation/backend_service/records.bal

```diff
@@ -0,0 +1,312 @@
+            value: re `^\d+`,
+            message: "Only numeric values allowed for PointOfServiceEntryMode"
+        }
+    }
+    string PointOfServiceEntryMode?;
+    @constraint:String {
+         pattern: {
+            value: re `^\d+`,
+            message: "Only numeric values allowed for PointOfServiceConditionCode"
+        }
+    }
+    string PointOfServiceConditionCode;
+    @constraint:String {
+         pattern: {
+            value: re `^\d+`,
+            message: "Only numeric values allowed for AmountTransactionFee"
+        }
+    }
+    string AmountTransactionFee?;
+    @constraint:String {
+         pattern: {
+            value: re `^\d+`,
+            message: "Only numeric values allowed for AcquiringInstitutionIdentificationCode"
+        }
+    }
+    string AcquiringInstitutionIdentificationCode;
+    @constraint:String {
+         pattern: {
+            value: re `^[a-zA-Z0-9]+$`,
+            message: "Only alpha numeric values allowed for RetrievalReferenceNumber"
+        }
+    }
+    string RetrievalReferenceNumber;
+    string CardAcceptorTerminalID?;
+    string CardAcceptorIDCode?;
+    string CardAcceptorNameLocation?;
+    @constraint:String {
+         pattern: {
+            value: re `^\d+`,
+            message: "Only numeric values allowed for CurrencyCodeTransaction"
+        }
+    }
+    string CurrencyCodeTransaction;
+    @constraint:String {
+        pattern: {
+            value: re `[0-1]+`,
+            message: "Only binary data allowed for IntegratedCircuitCardSystemRelatedData"
+        }
+    }
+    string IntegratedCircuitCardSystemRelatedData?;
+    string AccountIdentification1?;
+    string AccountIdentification2?;
+    string EftTlvData?; //todo - remove optional
+    @constraint:String {
+        pattern: {
+            value: re `[0-1]+`,
+            message: "Only binary data allowed for CardIssuerReferenceData"
+        }
+    }
+    string MessageAuthenticationCode?;
+    @constraint:String {
+        pattern: {
+            value: re `^\d+`,
+            message: "Only numeric values allowed for SettlementDate"
+        }
+    }
+    string SettlementDate?;
+    @constraint:String {
+        pattern: {
+            value: re `^\d+`,
+            message: "Only numeric values allowed for DateCapture"
+        }
+    }
+    string DateCapture?;
+    @constraint:String {
+        pattern: {
+            value: re `[0-1]+`,
+            message: "Only binary data allowed for PinData"
+        }
+    }
+    string PinData?;
+    string Track2Data?;
+    string ApplicationPANSequenceNumber?;
+    string CurrencyCodeCardholderBilling?;
+    string PersonalIdentificationNumberData?;
+    string ExpirationDate?;
+    string PointOfServiceCaptureCode?;
+    string CardSequenceNumber?;
+    string ForwardingInstitutionIdentificationCode?;
+|};
+
+public type MTI_0100 record {|
+    string MTI = "0100";
+    @constraint:String {
+         pattern: {
+            value: re `^\d+`,
```

Comment:
Is it possible to get these values from constants?

Reply:
seems it's not possible to define this as a constant